### PR TITLE
Hide battery module display for devices without battery

### PR DESCRIPTION
**Pull request by XaoflySho**
Hide battery module display for devices without battery.

### CODE REVIEW

File: eul/Schema/EulComponent.swift [hunk 1]
```diff
@@ -38,6 +38,14 @@ enum EulComponent: String, CaseIterable, Identifiable {
     case Memory
     case Battery
     case Network
+
+    static var allCases: [EulComponent] {
+        if BatteryStore().valid {
```
Comment (gao-sun) [inline eul/Schema/EulComponent.swift:43]: 1. no need for constructing a new instance here? just `BatteryStore.shared`
2. will it better to append battery component directly if needed, e.g.:

```swift
let components: [EulComponent] = [.CPU, .Fan, .Memory, .Network]
return BatteryStore.shared.isValid ? components + [.Battery] : components
```

File: eul/Store/BatteryStore.swift [hunk 2]
```diff
@@ -13,10 +13,11 @@ class BatteryStore: ObservableObject, Refreshable {
     static let shared = BatteryStore()
 
     private var battery = Battery()
-    private let opened = false
 
     var io = Info.Battery()
 
+    @Published var valid = true
```
Comment (gao-sun) [inline eul/Store/BatteryStore.swift:19]: maybe isValid?

File: eul/Store/BatteryStore.swift [hunk 3]
```diff
@@ -52,5 +56,6 @@ class BatteryStore: ObservableObject, Refreshable {
 
     init() {
         initObserver(for: .StoreShouldRefresh)
+        refresh()
```
Comment (gao-sun) [inline eul/Store/BatteryStore.swift:59]: i think `.StoreShouldRefresh` will be called once the app finished launching, see `AppDelegate.swift:53`

**Review by gao-sun — COMMENTED** [review]
thank you for contributing! overall looks great, and I left several comments. let me know if you have any questions.

File: eul/Schema/EulComponent.swift (hunk 1 above)
Comment (XaoflySho) [reply eul/Schema/EulComponent.swift:43]: Great advice!

File: eul/Store/BatteryStore.swift (hunk 2 above)
Comment (XaoflySho) [reply eul/Store/BatteryStore.swift:19]: I think your suggestion is good.

File: eul/Store/BatteryStore.swift (hunk 3 above)
Comment (XaoflySho) [reply eul/Store/BatteryStore.swift:59]: The call of `.StoreShouldRefresh` seems to be slower, and he does not guarantee that the `valid` is correct when using `BatteryStore`. Calling refresh manually is only one way to solve the problem, there may be a better way.

Comment (gao-sun) [reply eul/Store/BatteryStore.swift:59]: reasonable. it's cool to keep as it is for now

**Review by gao-sun — APPROVED** [review]
LGTM

**Pull request merged**